src/contracts/middleware/AvalancheL1Middleware.sol

```
@@ -594,34 +594,60 @@ contract AvalancheL1Middleware is IAvalancheL1Middleware, CollateralClassRegistr
         _initializeValidatorStakeUpdate(msg.sender, validationID, stakeAmount);
     }
 
-    /**
-     * @inheritdoc IAvalancheL1Middleware
-     */
-    function completeValidatorRegistration(
-        address operator,
-        bytes32 nodeId,
+
+
+    function completeValidatorRemoval(
         uint32 messageIndex
     ) external {
         _updateGlobalNodeStakeOncePerEpoch();
-        _completeValidatorRegistration(operator, nodeId, messageIndex);
+        _completeValidatorRemoval(messageIndex);
     }
 
-    /**
-     * @inheritdoc IAvalancheL1Middleware
-     */
-    function completeStakeUpdate(
-        bytes32 nodeId,
-        uint32 messageIndex
-    ) external {
+    // --- Permissionless completes (messageIndex-only) ---
+
+    function completeValidatorRegistration(uint32 messageIndex) external {
         _updateGlobalNodeStakeOncePerEpoch();
-        _completeStakeUpdate(msg.sender, nodeId, messageIndex);
+
+        bytes32 vid = balancerValidatorManager.completeValidatorRegistration(messageIndex);
+
+        // Only act for validators owned by this module
+        if (balancerValidatorManager.getValidatorSecurityModule(vid) != address(this)) {
+            return;
```

Should probably throw an exception no?

---

Actually there's no code after so I'm confused